dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -239,71 +157,45 @@ internal class RumSessionScope(
             else -> Datadog.startupTimeNs
         }
     }
-
-    @Synchronized
-    private fun updateSessionIdIfNeeded() {
+*/
+    private fun updateSession(event: RumRawEvent) {
         val nanoTime = System.nanoTime()
         val isNewSession = sessionId == RumContext.NULL_UUID
-        val sessionLength = nanoTime - sessionStartNs.get()
-        val duration = nanoTime - lastUserInteractionNs.get()
-        val isInactiveSession = duration >= sessionInactivityNanos
-        val isLongSession = sessionLength >= sessionMaxDurationNanos
 
-        if (isNewSession || isInactiveSession || isLongSession) {
-            keepSession = (random.nextFloat() * 100f) < samplingRate
-            sessionStartNs.set(nanoTime)
-            sessionId = UUID.randomUUID().toString()
-            sessionListener?.onSessionStarted(sessionId, !keepSession)
+        val timeSinceLastInteractionNs = nanoTime - lastUserInteractionNs.get()
+        val isExpired = timeSinceLastInteractionNs >= sessionInactivityNanos
+        val timeSinceSessionStartNs = nanoTime - sessionStartNs.get()
+        val isTimedOut = timeSinceSessionStartNs >= sessionMaxDurationNanos
+
+        val isInteraction = (event is RumRawEvent.StartView) || (event is RumRawEvent.StartAction)
+
+        if (isInteraction) {
+            if (isNewSession || isExpired || isTimedOut) {
+                renewSession(nanoTime)
+            }
+            lastUserInteractionNs.set(nanoTime)
+        } else {
+            if (isExpired) {
+                sessionState = State.EXPIRED
```

Comment:
it means we kind of reduce the value of the background tracking? Because for background events it won't be any interactions, so any new session created as a result of the background event (say network call) will expire quickly and events coming later won't be sent.